src/proofs/Term.cpp

```diff
@@ -192,12 +293,12 @@ std::shared_ptr<Term> RemoveUnusedVisitor::visit(Quant * term) {
 
     for (std::size_t i = 0; i < vars.size(); i++) {
         auto var = vars[i];
-        for (auto varInUse : varsInUse) {
+        for (auto const & varInUse : varsInUse) {
             if (var->printTerm() == varInUse) { inUse = true; }
         }
         if (!inUse) {
-            vars.erase(vars.begin() + i);
-            sorts.erase(sorts.begin() + i);
+            vars.erase(vars.begin() + int(i));
+            sorts.erase(sorts.begin() + int(i));
         }
         inUse = false;
     }
```

Comment:
This loop does not really work correctly, because if there are two consecutive unused variables, this loop will skip the second one.

Here is an example that generated invalid proof according to carcara:
```
(set-logic HORN)
(declare-fun P (Int) Bool)

(assert (forall ((x Int) (unused Int) (unused2 Bool)) (=> (= x 0) (P x))))

(assert (forall ((x Int)) (=> (P x) false)))


(check-sat)
(exit)
```

If `inUse` is `false`, `i` should not be incremented, because `erase` moves the rest of the elements one place lower.